File: urbanairship-core/src/main/java/com/urbanairship/actions/ActionRunRequestExtender.java
/* Copyright Airship and Contributors */

package com.urbanairship.actions;

import androidx.annotation.NonNull;

/**
 * Extends an action run request.
 */
public interface ActionRunRequestExtender {

    /**
     * Extends an action run request.
     *
     * @param request The request.
     * @return The extended request.
     */
    @NonNull
    ActionRunRequest extend(@NonNull ActionRunRequest request);

}

File: urbanairship-core/src/main/java/com/urbanairship/actions/ActionRunRequestExtender.kt
/* Copyright Airship and Contributors */
package com.urbanairship.actions

/**
 * Extends an action run request.
 */
public interface ActionRunRequestExtender {

    /**
     * Extends an action run request.
     *
     * @param request The request.
     * @return The extended request.
     */
    public fun extend(request: ActionRunRequest): ActionRunRequest
}
